accepted/00000-auto-generate-bootstrap-repositories.md

```diff
@@ -0,0 +1,71 @@
+
+Users very often do not know this, or forget to create them when they synchronize a new product. Also the bootstrap repository
+is only updated when the user execute the generator script again.
+
+This RFC should describe how to change the current tools to automatically 
+execute the generation of bootstrap repositories after
+a new product was added and after a channel changed when a maintenance update was synced to Uyuni Server.
+
+# Detailed design
+[design]: #detailed-design
+
+The currently exiting tool `mgr-create-bootstrap-repo` list all available bootstrap repositories it could generate.
+This list depends on the products the user has synchronized.
+
+Step 1: provide an option to enable an automatic mode which iterate over all available bootstrap repositories and generate them
+when needed. As we have sometimes have multiple options for the same local "path" we should iterate over the paths to minimize
+the number of runs.
+
+Step 2: add a log file to `mgr-create-bootstrap-repo`. Currently it has no log file. But when this tool is running in the background
+we need the possibility to debug problems. All messages should go to a log file.
+
+Step 3: let `mgr-create-bootstrap-repo` use a configuration file. All commandline options should be possible to set via
+configuration file. This allows to influence the behavior even in automatic mode when it is not possible to change the commandline
+options.
+
+To trigger the re-generation we implement the following algorithm:
+
+- Create a taskomatic job running every 15 minutes and call `mgr-create-bootstrap-repo --auto`
+- It calculate all channels which are needed to create the bootstrap repository for every available OS
+- It compare the `last_modified` timestamps of the channels (DB) with the `modified date` of the main metadata file
+  (repomd.xml or Release)
+- if all `last_modified` timestamps are newer then the timestamp of the main metadata file, re-generate the bootstrap repository
+- if any `last_modified` timestamps are newer then the timestamp of the main metadata file and the `modified date` of the main metadata
+  file is older than 26 hours, re-generate the bootstrap repository. This is for the case a channel is not synced. It can happen on
+  custom repositories needed for bootstrap repositories.
+
+Provide a `--force` option to enforce recreation of all bootstrap repositories without checking if needed. Mostly for manual usage.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+We maybe 15 minutes late with the bootstrap repository or maybe longer when a channel failed to sync or does not sync at all.
+
+# Alternatives
+[alternatives]: #alternatives
+
+1. Execute the `mgr-create-bootstrap-repo` one time per day in the morning via taskomatic expecting the nightly reposync is finished.
+
+2. Every finished repo sync trigger a `mgr-create-bootstrap-repo --for_updated_channel <label>`. It check if this channel is used
```

Comment:
It sounds like this is the same proposal I mentioned above. What are the pros/cons of this one?

Reply:
The difference is, that this is triggered by reposync itself and it was not planned to check for changes.
But you are right, they are pretty similar.

We could remove sheduling via taskomatic or systemd and call it directly from reposync. This would
remove the need of an extra trigger job.
What do you think?

Reply:
I think I like it. In the end, bootstrap repository generation should be a sequential step after reposync, so if we can avoid the trigger job and pose no disadvantages, I'm all for it.

Reply:
Agreed. I change the RFC